### PR TITLE
CREATE TABLE and CREATE INDEX storage parameters/ON COMMIT PRESERVE ROW

### LINKED ISSUE #8160: sql: allow storage_params with CREATE INDEX syntax
PR: https://github.com/cockroachdb/cockroach/pull/52640

From release notes:
> Added support for the use of `CREATE INDEX ... WITH ...` syntax. [#52640][#52640] {% comment %}doc{% endcomment %}

### LINKED ISSUE #8162: sql: enable autovacuum_enabled storage param setting
PR: https://github.com/cockroachdb/cockroach/pull/52652

From release notes:
> Added support for parsing the syntax `CREATE TABLE ... WITH (autovacuum_enabled = bool)`, which results in a no-op. [#52652][#52652] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #8162. 
Fixes #8160.

### CODE REVIEW

File: v20.2/create-table-as.md [hunk 1]
```diff
@@ -79,6 +85,8 @@ table td:first-child {
  `create_as_constraint_def` | An optional [primary key constraint](primary-key.html).
  `select_stmt` | A [selection query](selection-queries.html) to provide the data.
  `opt_persistence_temp_table` |  Defines the table as a session-scoped temporary table. For more information, see [Temporary Tables](temporary-tables.html).<br><br>Note that the `LOCAL`, `GLOBAL`, and `UNLOGGED` options are no-ops, allowed by the parser for PostgresSQL compatibility.<br><br>**Support for temporary tables is [experimental](experimental-features.html#temporary-objects)**.
+ `opt_with_storage_parameter_list` | This clause is a no-op, allowed by the parser for PostgresSQL compatibility. Allowed parameters include `fillfactor`, `s2_max_level`, `s2_level_mod`, `s2_max_cells`, `geometry_min_x`, `geometry_max_x`, `geometry_min_y`, and `geometry_max_y`.
```
Comment (otan) [inline v20.2/create-table-as.md:88]: not entirely a no-op -- the `s2_*` commands configure geospatial indexes

File: v20.2/create-table-as.md [hunk 2]
```diff
@@ -79,6 +85,8 @@ table td:first-child {
...
+ `ON COMMIT PRESERVE ROWS` | This clause is a no-op, allowed by the parser for PostgresSQL compatibility. CockroachDB only supports session-scoped temp tables, and does not support the clauses `ON COMMIT DELETE ROWS` and `ON COMMIT DROP`, which are used to define transaction-scoped temp tables in PostgreSQL.
```
Comment (otan) [inline v20.2/create-table-as.md:89]: temp or temporary?

Comment (ericharmeling) [reply v20.2/create-table-as.md:89]: Updated to read "temporary tables"... but I did not know that there was a difference. Is there a difference?

File: v20.2/create-table-as.md (hunk 1 above)
Comment (ericharmeling) [reply v20.2/create-table-as.md:88]: Oh wow you are totally right. Updated this.

**Comment by ericharmeling** [comment]
@otan TFTR! Please re-review latest commit.

File: v20.2/create-table-as.md (hunk 2 above)
Comment (otan) [reply v20.2/create-table-as.md]: nah just wondering about using full words or not

**Review by otan — APPROVED** [review]

**Review by lnhsingh — COMMENTED** [review]
Question on whether the `ON COMMIT PRESERVE ROWS` definitions should have a "New in 20.2" tag, but otherwise LGTM

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8723#-:-MK_O7UM4zSMJ5el_L7C:biqsg3j)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @lnhsingh, and @otan)

___
*[v20.2/create-table.md, line 107 at r2](https://reviewable.io/reviews/cockroachdb/docs/8723#-MK_O-mSAzHcleoYDygi:-MK_O-mSAzHcleoYDygj:b-gt9zb5) ([raw file](https://github.com/cockroachdb/docs/blob/b29e799ed6933d0ac2d2521ae7389b1a3177c3d5/v20.2/create-table.md#L107)):*
> ```Markdown
> `opt_where_clause` | <span class="version-tag">New in v20.2:</span> An optional `WHERE` clause that defines the predicate boolean expression of a [partial index](partial-indexes.html).
> `opt_with_storage_parameter_list` | <span class="version-tag">New in v20.2:</span> A comma-separated list of [spatial index tuning parameters](spatial-indexes.html#index-tuning-parameters). Supported parameters include `fillfactor`, `s2_max_level`, `s2_level_mod`, `s2_max_cells`, `geometry_min_x`, `geometry_max_x`, `geometry_min_y`, and `geometry_max_y`. The `fillfactor` parameter is a no-op, allowed for PostgreSQL-compatibility.<br><br>For details, see [Spatial index tuning parameters](spatial-indexes.html#index-tuning-parameters). For an example, see [Create a spatial index that uses all of the tuning parameters](spatial-indexes.html#create-a-spatial-index-that-uses-all-of-the-tuning-parameters).
> `ON COMMIT PRESERVE ROWS` | This clause is a no-op, allowed by the parser for PostgresSQL compatibility. CockroachDB only supports session-scoped [temporary tables](temporary-tables.html), and does not support the clauses `ON COMMIT DELETE ROWS` and `ON COMMIT DROP`, which are used to define transaction-scoped temporary tables in PostgreSQL.
> ```

Should all of the `ON COMMIT PRESERVE ROWS` definitions have a "New in 20.2" tag?

**Review by ericharmeling — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/8723)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling, @lnhsingh, and @otan)

___
*[v20.2/create-table.md, line 107 at r2](https://reviewable.io/reviews/cockroachdb/docs/8723#-MK_O-mSAzHcleoYDygi:-MKaFUxj-NdC6e4DfNW2:brsapzd) ([raw file](https://github.com/cockroachdb/docs/blob/b29e799ed6933d0ac2d2521ae7389b1a3177c3d5/v20.2/create-table.md#L107)):*
<details><summary><i>Previously, lnhsingh (Lauren Hirata Singh) wrote…</i></summary><blockquote>

Should all of the `ON COMMIT PRESERVE ROWS` definitions have a "New in 20.2" tag?
</blockquote></details>

This was actually introduced in 20.1, but the generator wasn't picking up this clause, because the clause's field in the grammar file was empty (see https://github.com/cockroachdb/cockroach/pull/55832/files#diff-570c32ab78bd63a6e239714da4bcbaed11c42a87d8b0f44552a646875ba9b317R5580). We document this clause on the temp tables page (https://www.cockroachlabs.com/docs/v20.1/temporary-tables#details), so I'm not sure it's worth the effort checking out the v20.1 source and regenerating the `CREATE TABLE` diagram.

**Pull request merged**